FEC-9910 Kaltura Player - Basic Flow

x-NR-x opened the pull request:
A basic player with kava impression only.

Solves FEC-9910

Replaces (#2) cause the branch was renamed to FEC instead of FEM

x-NR-x committed:
FEM-2604 KalturaPlayer remodeling. Starting Basic Flow.

x-NR-x committed:
FEM-2604 WIP - Basic Flow

x-NR-x committed:
FEC-9910 Basic Flow.

x-NR-x committed:
Fixed Kava plugin not initialized.
Made public some properties.

x-NR-x committed:
Removed objc.

x-NR-x committed:
Added iOSTestApp, updated travis with the travis-build.

x-NR-x committed:
Updated the Readme.

x-NR-x committed:
Removed media data from the BasicPlayerOptions class.
Added a PKMediaEntry property to be set.
Added a setupMediaEntry function in the BasicPlayer to set the mediaEntry from parameters.
Removed registration to the Kava plugin in the BasicPlayer, only an impression is sent.
Removed the prepare from the KalturaBasicPlayer class into the KalturaPlayer class.
Changed the setPlayerView function to a kalturaPlayerView property.
Added preload, autoplay and startTime cases.
Renamed player property to pkPlayer for better understanding.

x-NR-x committed:
FEC-9959 Added an option to send the mediaType in the setupMediaEntry function.
Set the startTime to be optional, and creating the MediaConfig with the startTime only if it was set.
Removed the play after prepare in case of autoPlay and added an observer to the PlayerEvent.CanPlay and set it there, otherwise the currentItem is not set and in case of a live media it won't seek to the live edge.

x-NR-x committed:
Removed calling play only after canPlay event, this was not the issue and will interfere with play immediately functionality.

x-NR-x committed:
FEC-9910 Added an option to update the PlayerOptions for the player.

x-NR-x committed:
Removed the old example, the samples can be found in the samples repo.

kaltura-hooks commented:
Hi @x-NR-x,
Thank you for contributing this pull request!
Please sign the Kaltura CLA so we can review and merge your contribution.
Learn more at http://bit.ly/KalturaContrib

x-NR-x committed:
Updated and added missing documentation.

x-NR-x committed:
FEC-9910 Removed the 'from' naming of the id param.

x-NR-x committed:
Renamed kalturaPlayerView property to view.
Removed check on the didSet of the mediaEntry for same object.

x-NR-x committed:
FEC-9910 Added MediaOptions.
Moved startTime from the player options to the media options, updated relevant code accordingly.
Set the mediaEntry in the KalturaPlayer as internal and added a setMedia function in the BasicKalturaPlayer which can be set with the media options as well.
Updated the KalturaBasicPlayer initializer naming for a cleaner reading approach.
